crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -513,14 +530,32 @@ fn strip_comment_prefix(comment_text: &str) -> FormatResult<&str> {
 /// ```
 ///
 /// This builder will insert two empty lines before the comment.
+///
+/// # Preview
+///
+/// For preview style, this builder will insert a single empty line after a
+/// class definition in a stub file.
+///
+/// For example, given:
+/// ```python
+/// class Foo:
+///     pass
+/// # comment
+/// ```
+///
+/// This builder will insert a single empty line before the comment.
 pub(crate) fn empty_lines_before_trailing_comments<'a>(
     f: &PyFormatter,
     comments: &'a [SourceComment],
+    node_kind: NodeKind,
 ) -> FormatEmptyLinesBeforeTrailingComments<'a> {
     // Black has different rules for stub vs. non-stub and top level vs. indented
     let empty_lines = match (f.options().source_type(), f.context().node_level()) {
         (PySourceType::Stub, NodeLevel::TopLevel(_)) => 1,
-        (PySourceType::Stub, _) => 0,
+        (PySourceType::Stub, _) => u32::from(
+            is_blank_line_after_nested_stub_class_enabled(f.context())
+                && node_kind == NodeKind::StmtClassDef,
+        ),
```

Comment:
Logic for when a class is followed by a trailing comment.

Reply:
I did update the docs for the function :)

https://github.com/astral-sh/ruff/blob/a768fc35ca1f73012ae9d76cba2eedda95c9db1e/crates/ruff_python_formatter/src/comments/format.rs#L534-L546

Is this what you meant?